core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -372,8 +376,27 @@ private[spark] class SparkSubmit extends Logging {
       localJars = Option(args.jars).map {
         downloadFileList(_, targetDir, sparkConf, hadoopConf, secMgr)
       }.orNull
-      localPyFiles = Option(args.pyFiles).map {
-        downloadFileList(_, targetDir, sparkConf, hadoopConf, secMgr)
+      localPyFiles = Option(args.pyFiles).map { pyFiles =>
+        if (isClientPythonSubmit) {
```

Comment:
Couldn't this logic be in `PythonRunner`? That's basically what SparkSubmit runs when the conditions you use to create `isClientPythonSubmit` are met.

This class is already pretty hard to navigate, it'd be better to avoid adding more special cases to it.

Reply:
Yup, it can be. Will try.

Reply:
Agreed with @vanzin , we can move this logic to python related code.